cellmlmanip/units.py

```diff
@@ -278,6 +286,37 @@ def convert(self, quantity, unit):
         assert isinstance(unit, self._registry.Unit)
         return quantity.to(unit)
 
+    def add_conversion_rule(self, from_unit, to_unit, rule):
+        """Adds complex conversion rule for converting between incompatible units.
+
+        For example:
+        uA = units.add_unit('uA', 'ampere * 1e-6')
+        pA = units.add_unit('pA', 'ampere * 1e-12')
+        uF = units.add_unit('uF', 'farad * 1e-6')
+        pF = units.add_unit('pF', 'farad * 1e-12')
+        cm2 = units.add_unit('cm2', 'centimetre ** 2')
+        uA_per_cm2 = uA_per_cm2 = units.add_unit('uA_per_cm2', 'uA / cm2')
+        uF_per_cm2 = units.add_unit('uF_per_cm2', 'uF / cm2')
+        A_per_F = units.add_unit('A_per_F', 'ampere / farad')
+
+        Cm = units.Quantity(12, pF)
+        Cs = units.Quantity(1.1, uF_per_cm2)
+
+        c.add_transformation(uA, uA_per_cm2, lambda ureg, rhs: rhs * Cs / Cm)
+        c.add_transformation(uA_per_cm2, A_per_F, lambda ureg, rhs: rhs / Cs)
+
+        # Test
+        x = r.Quantity(1, pA)
+        print(x.to(A_per_F))
```

Comment:
This won't format nicely, and also doesn't use the new method! I think something like the following:
```suggestion
        For example::

            units = UnitStore()
            uA = units.add_unit('uA', 'ampere * 1e-6')
            pA = units.add_unit('pA', 'ampere * 1e-12')
            uF = units.add_unit('uF', 'farad * 1e-6')
            pF = units.add_unit('pF', 'farad * 1e-12')
            cm2 = units.add_unit('cm2', 'centimetre ** 2')
            uA_per_cm2 = uA_per_cm2 = units.add_unit('uA_per_cm2', 'uA / cm2')
            uF_per_cm2 = units.add_unit('uF_per_cm2', 'uF / cm2')
            A_per_F = units.add_unit('A_per_F', 'ampere / farad')

            Cm = units.Quantity(12, pF)
            Cs = units.Quantity(1.1, uF_per_cm2)

            units.add_conversion_rule(uA, uA_per_cm2, lambda ureg, rhs: rhs * Cs / Cm)
            units.add_conversion_rule(uA_per_cm2, A_per_F, lambda ureg, rhs: rhs / Cs)

            # Test
            print(units.convert(units.Quantity(1, pA), A_per_F))
            print(units.convert(units.Quantity(1, uA_per_cm2), A_per_F))
```